### PR TITLE
Auto generated api documentation

### PR DESCRIPTION
**Pull request by jpatters**
This builds and publishes auto generated api documentation.

### CODE REVIEW

File: typedoc.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ limitations under the License.
 
 module.exports = packageJson => {
   const gitRevision = `${packageJson.name}@${packageJson.version}`
+  const outDir = `${__dirname}/docs/${gitRevision}`
```
Comment (ncphillips) [inline typedoc.js]: This is nice 👍

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '10.x'
```
Comment (ncphillips) [inline .github/workflows/docs.yml]: Double check if this is needed. IIRC It's in the build script for running on macos

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+      - name: Install
+        run: |
+          npm ci
+          npm run bs
```
Comment (ncphillips) [inline .github/workflows/docs.yml]: I could see this not being necessary

File: .github/workflows/docs.yml (hunk 2 above)
Comment (jpatters) [reply .github/workflows/docs.yml]: It might not be necessary, but I would rather be explicit about the version of node that it is using.

**Comment by ncphillips** [comment]
@all-contributors please add @jpatters for infra and docs

**Pull request merged**